modules/internal/ChapelBase.chpl

```diff
@@ -103,7 +103,25 @@ module ChapelBase {
     __primitive("=", a, nil);
   }
 
+  inline proc =(ref a: void, b: ?t) where t != void {
+    compilerError("a void variable cannot be assigned");
+  }
+
+  inline proc =(ref a: ?t, b: void) where t != void {
+    compilerError("cannot assign void to a variable of non-void type");
+  }
 
+  inline proc =(ref a: void, b: void) param {
+    return _void;
```

Comment:
Most =() overloads don't return anything -- does this one need to return something for some reason?  I was expecting an empty body here.